test/host/socket/wasi_socket.cpp

```diff
@@ -124,6 +127,13 @@ void allocateAddrinfoArray(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
     }
   }
 }
+
+template <typename T>
+T getValue(WasmEdge::Runtime::Instance::MemoryInstance &MemInst, uint32_t Ptr) {
+  T Value;
+  MemInst.loadValue(Value, Ptr);
+  return Value;
+}
```

Comment:
Is there any benefit to having this helper function?

Reply:
Most of them are something like this: `*MemInst.getPointer<const uint32_t *>(0)`. It avoids calling a template `getValue` function and reduces the calling frame cost. I didn't find the `extra` variables. Instead of your new method, it will always create a new `T Value` when calling this function.